hack/mock-unittest-data/gomod.py

```diff
@@ -0,0 +1,175 @@
+        fpath.write_text(data)
+
+
+def update_workspaces(run, run_redirect, pandemo_dir, mocked_data_dir_abspath, mocked_data_dir):
+    with open(f"{mocked_data_dir_abspath}/workspaces/go_work.json") as f:
+        d_paths = [v['DiskPath'] for v in json.loads(f.read())['Use']]
+        paths = [Path(f"{pandemo_dir}")/p.removeprefix("./") for p in d_paths if p != '.']
+
+    for p in paths:
+        print(f"preparing per-workspace directory {p}")
+        mkdir_p(f"{mocked_data_dir_abspath}/workspaces/{p}")
+        if (gosum:=Path(p)/"go.sum").exists():
+            print(f"generating {mocked_data_dir}/workspaces/{p.name}/go.sum")
+            copyfile(gosum, Path(f"{mocked_data_dir_abspath}/workspaces/{p.name}/go.sum"))
+        inject_cwd(run, p)
+        run_redirect(
+            where=f"workspaces/{p.name}/go_list_deps_threedot.json",
+            what="go list -deps -json=ImportPath,Module,Standard,Deps ./..."
+        )
+
+    inject_cwd(run, pandemo_dir)
+
+
+def warn_about_potential_manual_intervention(mocked_data_dir):
+    non_vendored_diff = subprocess.run(
+        ["git", "diff", "--", f"{mocked_data_dir}/non-vendored"],
+        capture_output=True, text=True, check=True
+    ).stdout
+    vendored_diff = subprocess.run(
+        ["git", "diff", "--", f"{mocked_data_dir}/vendored"],
+        capture_output=True, text=True, check=True
+    ).stdout
+    if vendored_diff or non_vendored_diff:
+        nonvendor = f"{mocked_data_dir}/expected-results/resolve_gomod.json"
+        vendor = f"{mocked_data_dir}/expected-results/resolve_gomod_vendored.json"
+        banner = f"""The mock data changed => the expected unit test results may change.
+            The following files may need to be adjusted manually:
+            {nonvendor if non_vendored_diff else ""}
+            {vendor if vendored_diff else ""}
+        """
+        print_banner(banner)
+
+
+def main() -> None:
+    print_banner("Generating mock data for gomod unit tests")
+    mocked_data_dir = Path(argv[1] if len(argv) > 1 else "tests/unit/data/gomod-mocks")
+    mocked_data_dir_abspath, tmpdir = generate_auxiliary_dirs(mocked_data_dir)
+    with pandemonium_dir(tmpdir) as pandemo_dir:
+        params = {"env": environ | {"GOMODCACHE": tmpdir/"hermeto-mock-gomodcache"}, "check": True}
+        run = partial(subprocess.run, **params)
+        run(_c(f"git clone https://github.com/cachito-testing/gomod-pandemonium {pandemo_dir}"))
+        inject_cwd(run, pandemo_dir)
+        run(_c("git switch go-1.22-workspaces"))
+        print(f"generating {mocked_data_dir}/workspaces/go.sum")
+        copyfile(pandemo_dir/"go.sum", Path(f"{mocked_data_dir_abspath}/workspaces/go.sum"))
+        run_redirect = partial(
+            run_with_redirection, run=run, mocked_data_dir=mocked_data_dir,
+            mocked_data_dir_abspath=mocked_data_dir_abspath
+        )
+        run_redirect(where="workspaces/go_list_modules.json", what="go list -m -json")
```

Comment:
I guess I'd prefer a few for loops instead of a partial, but it's readable, so let's keep `run_redirect` as you have it.